--- Eigen/src/QR/Tridiagonalization.h	2010-07-22 14:45:32.000000000 -0700
+++ Eigen/src/QR/Tridiagonalization.h	2010-07-22 14:46:26.000000000 -0700
@@ -391,7 +391,7 @@
 {
   diag[0] = ei_real(mat(0,0));
   RealScalar v1norm2 = ei_abs2(mat(0,2));
-  if (ei_isMuchSmallerThan(v1norm2, RealScalar(1)))
+  if (v1norm2==RealScalar(0))
   {
     diag[1] = ei_real(mat(1,1));
     diag[2] = ei_real(mat(2,2));
