Index: src/LinearMath/btMatrix3x3.h
===================================================================
--- src/LinearMath/btMatrix3x3.h	(revision 2038)
+++ src/LinearMath/btMatrix3x3.h	(working copy)
@@ -192,6 +192,16 @@
 			-sj,      cj * si,      cj * ci);
 	}
 
+	/** @brief Set the matrix using RPY about XYZ fixed axes
+	 * @param roll Roll about X axis
+         * @param pitch Pitch around Y axis
+         * @param yaw Yaw aboud Z axis
+         * 
+	 **/
+	void setRPY(btScalar roll, btScalar pitch,btScalar yaw) { 
+               setEulerYPR(yaw, pitch, roll);
+	}
+
 	/**@brief Set the matrix to the identity */
 	void setIdentity()
 	{ 
@@ -361,6 +371,16 @@
 		}
 	}
 
+	/**@brief Get the matrix represented as roll pitch and yaw about fixed axes XYZ
+	* @param roll around X axis 
+	* @param pitch Pitch around Y axis
+	* @param yaw Yaw around Z axis
+	* @param solution_number Which solution of two possible solutions ( 1 or 2) are possible values*/	
+	void getRPY(btScalar& roll, btScalar& pitch, btScalar& yaw, unsigned int solution_number = 1) const
+	{
+	getEulerYPR(yaw, pitch, roll, solution_number);
+	}
+
 	/**@brief Create a scaled copy of the matrix 
 	* @param s Scaling vector The elements of the vector will scale each column */
 
