Index: CMakeLists.txt
===================================================================
--- CMakeLists.txt	(revision 1779)
+++ CMakeLists.txt	(working copy)
@@ -64,7 +64,7 @@
   SUBDIRS(Demos)
 ENDIF(BUILD_DEMOS)
 
-OPTION(BUILD_EXTRAS "Set when you want to build the extras" ON)
+OPTION(BUILD_EXTRAS "Set when you want to build the extras" OFF)
 IF(BUILD_EXTRAS)
   SUBDIRS(Extras)
 ENDIF(BUILD_EXTRAS)
