Index: src/LinearMath/btQuaternion.h
===================================================================
--- src/LinearMath/btQuaternion.h	(revision 1780)
+++ src/LinearMath/btQuaternion.h	(working copy)
@@ -200,8 +200,11 @@
 	{
 		btScalar s = btSqrt(length2() * q.length2());
 		btAssert(s != btScalar(0.0));
-		return btAcos(dot(q) / s);
+               if (dot(q) < 0) // Take care of long angle case see http://en.wikipedia.org/wiki/Slerp
+                     return btAcos(dot(-q) / s) * btScalar(2.0);
+               else 
+                     return btAcos(dot(q) / s) * btScalar(2.0);
 	}
   /**@brief Return the angle of rotation represented by this quaternion */
 	btScalar getAngle() const 
 	{
