Index: src/LinearMath/btQuaternion.h
===================================================================
--- src/LinearMath/btQuaternion.h	(revision 2038)
+++ src/LinearMath/btQuaternion.h	(working copy)
@@ -194,7 +202,7 @@
 	{
 		return *this / length();
 	} 
-  /**@brief Return the angle between this quaternion and the other 
+  /**@brief Return the ***half*** angle between this quaternion and the other 
    * @param q The other quaternion */
 	btScalar angle(const btQuaternion& q) const 
 	{
@@ -202,6 +210,17 @@
 		btAssert(s != btScalar(0.0));
 		return btAcos(dot(q) / s);
 	}
+	/**@brief Return the angle between this quaternion and the other along the shortest path
+	* @param q The other quaternion */
+	btScalar angleShortestPath(const btQuaternion& q) const 
+	{
+		btScalar s = btSqrt(length2() * q.length2());
+		btAssert(s != btScalar(0.0));
+		if (dot(q) < 0) // Take care of long angle case see http://en.wikipedia.org/wiki/Slerp
+			return btAcos(dot(-q) / s) * btScalar(2.0);
+		else 
+			return btAcos(dot(q) / s) * btScalar(2.0);
+	}
   /**@brief Return the angle of rotation represented by this quaternion */
 	btScalar getAngle() const 
 	{
@@ -209,6 +228,18 @@
 		return s;
 	}
 
+	/**@brief Return the angle of rotation represented by this quaternion along the shortest path*/
+	btScalar getAngleShortestPath() const 
+	{
+	btScalar s;
+		if (dot(*this) < 0)
+			s = btScalar(2.) * btAcos(m_floats[3]);
+		else
+			s = btScalar(2.) * btAcos(-m_floats[3]);
+
+		return s;
+	}
+
 	/**@brief Return the axis of the rotation represented by this quaternion */
 	btVector3 getAxis() const
 	{
@@ -279,7 +310,7 @@
    * Slerp interpolates assuming constant velocity.  */
 	btQuaternion slerp(const btQuaternion& q, const btScalar& t) const
 	{
-		btScalar theta = angle(q);
+          btScalar theta = angleShortestPath(q) / btScalar(2.0);
 		if (theta != btScalar(0.0))
 		{
 			btScalar d = btScalar(1.0) / btSin(theta);
@@ -366,13 +397,20 @@
 	return q.length(); 
 }
 
-/**@brief Return the angle between two quaternions*/
+/**@brief Return the ***half*** angle between two quaternions*/
 SIMD_FORCE_INLINE btScalar
 angle(const btQuaternion& q1, const btQuaternion& q2) 
 { 
 	return q1.angle(q2); 
 }
 
+/**@brief Return the shortest angle between two quaternions*/
+SIMD_FORCE_INLINE btScalar
+angleShortestPath(const btQuaternion& q1, const btQuaternion& q2) 
+{ 
+	return q1.angleShortestPath(q2); 
+}
+
 /**@brief Return the inverse of a quaternion*/
 SIMD_FORCE_INLINE btQuaternion
 inverse(const btQuaternion& q) 
