Index: src/bindings/python/CMakeLists.txt
===================================================================
--- src/bindings/python/CMakeLists.txt	(revision 31693)
+++ src/bindings/python/CMakeLists.txt	(working copy)
@@ -7,6 +7,21 @@
   ${CMAKE_SOURCE_DIR}/src
   ${EIGEN2_INCLUDE_DIR})
 
+
+# use, i.e. don't skip the full RPATH for the build tree
+SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+
+# when building, don't use the install RPATH already
+# (but later on when installing)
+SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE) 
+
+# the RPATH to be used when installing
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+
+# add the automatically determined parts of the RPATH
+# which point to directories outside the build tree to the install RPATH
+SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
 file(GLOB SIP_FILES "${CMAKE_CURRENT_SOURCE_DIR}/*.sip")
 set(SIP_INCLUDES ${SIP_FILES})
 add_sip_python_module(PyKDL PyKDL/PyKDL.sip orocos-kdl)
Index: config/FindSIPandPython.cmake
===================================================================
--- config/FindSIPandPython.cmake	(revision 31715)
+++ config/FindSIPandPython.cmake	(working copy)
@@ -16,16 +16,31 @@
   ENDIF (UNIX AND NOT APPLE)
   FIND_PACKAGE(SIP)
   #INCLUDE(${PROJ_SOURCE_DIR}/config/FindSIP.cmake)
+
   IF (SIP_FOUND)
     # check for SIP version
-    SET (SIP_MIN_VERSION "4.7.9")
-    IF (${SIP_VERSION} VERSION_EQUAL "${SIP_MIN_VERSION}" OR ${SIP_VERSION} VERSION_GREATER "${SIP_MIN_VERSION}")
-      SET (SIP_IS_GOOD TRUE)
-    ENDIF (${SIP_VERSION} VERSION_EQUAL "${SIP_MIN_VERSION}" OR ${SIP_VERSION} VERSION_GREATER "${SIP_MIN_VERSION}")
-    
-    IF (NOT SIP_IS_GOOD)
-      MESSAGE (STATUS "SIP is required in version 4.7.9 or later!")
-    ENDIF (NOT SIP_IS_GOOD)
+
+    # cmake 2.6 and greater
+    IF ("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" GREATER 2.4)
+      SET (SIP_MIN_VERSION "4.7.9")
+      IF (${SIP_VERSION} VERSION_EQUAL "${SIP_MIN_VERSION}" OR ${SIP_VERSION} VERSION_GREATER "${SIP_MIN_VERSION}")
+        SET (SIP_IS_GOOD TRUE)
+      ELSE (${SIP_VERSION} VERSION_EQUAL "${SIP_MIN_VERSION}" OR ${SIP_VERSION} VERSION_GREATER "${SIP_MIN_VERSION}")
+        MESSAGE("SIP is required in versin 4.7.9 or greater")
+      ENDIF (${SIP_VERSION} VERSION_EQUAL "${SIP_MIN_VERSION}" OR ${SIP_VERSION} VERSION_GREATER "${SIP_MIN_VERSION}")
+    # cmake 2.4
+    ELSE ("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" GREATER 2.4)
+      SET (SIP_MIN_VERSION 040703)
+      IF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
+        SET (SIP_IS_GOOD TRUE)
+      ELSE (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
+        MESSAGE("SIP is required in versin 4.7.3 or greater")
+      ENDIF (SIP_VERSION EQUAL "${SIP_MIN_VERSION}" OR SIP_VERSION GREATER "${SIP_MIN_VERSION}")
+    ENDIF ("${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION}" GREATER 2.4)
+    MESSAGE(${SIP_VERSION})
+    MESSAGE(${SIP_MIN_VERSION})
+    MESSAGE(${CMAKE_VERSION})
+  
   ELSE (SIP_FOUND)
     MESSAGE (STATUS "SIP not found!")
   ENDIF (SIP_FOUND)
