Index: src/bindings/python/CMakeLists.txt
===================================================================
--- src/bindings/python/CMakeLists.txt	(revision 31715)
+++ src/bindings/python/CMakeLists.txt	(working copy)
@@ -7,6 +7,21 @@
   ${CMAKE_SOURCE_DIR}/src
   ${EIGEN2_INCLUDE_DIR})
 
+
+# use, i.e. don't skip the full RPATH for the build tree
+SET(CMAKE_SKIP_BUILD_RPATH  FALSE)
+
+# when building, don't use the install RPATH already
+# (but later on when installing)
+SET(CMAKE_BUILD_WITH_INSTALL_RPATH FALSE) 
+
+# the RPATH to be used when installing
+SET(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_PREFIX}/lib")
+
+# add the automatically determined parts of the RPATH
+# which point to directories outside the build tree to the install RPATH
+SET(CMAKE_INSTALL_RPATH_USE_LINK_PATH TRUE)
+
 file(GLOB SIP_FILES "${CMAKE_CURRENT_SOURCE_DIR}/*.sip")
 set(SIP_INCLUDES ${SIP_FILES})
 add_sip_python_module(PyKDL PyKDL/PyKDL.sip orocos-kdl)


--- src/tree.cpp	2010-07-23 09:58:14.435307889 -0700
+++ src/tree.cpp	2009-08-14 00:54:19.000000000 -0700
@@ -142,9 +142,9 @@
         Frame f_tip = seg.pose(0.0).Inverse();
         Joint jnt = seg.getJoint();
         if (jnt.getType() == Joint::RotX || jnt.getType() == Joint::RotY || jnt.getType() == Joint::RotZ || jnt.getType() == Joint::RotAxis)
+	  jnt = Joint(jnt.getName(), f_tip*jnt.JointOrigin(), f_tip.M*(-jnt.JointAxis()), Joint::RotAxis);
+	else if (jnt.getType() == Joint::TransX || jnt.getType() == Joint::TransY || jnt.getType() == Joint::TransZ || jnt.getType() == Joint::TransAxis)
+	  jnt = Joint(jnt.getName(),f_tip*jnt.JointOrigin(), f_tip.M*(-jnt.JointAxis()), Joint::TransAxis);
-            jnt = Joint(jnt.getName(), f_tip*jnt.JointOrigin(), f_tip.M*jnt.JointAxis(), Joint::RotAxis);
-        else if (jnt.getType() == Joint::TransX || jnt.getType() == Joint::TransY || jnt.getType() == Joint::TransZ || jnt.getType() == Joint::TransAxis)
-            jnt = Joint(jnt.getName(),f_tip*jnt.JointOrigin(), f_tip.M*jnt.JointAxis(), Joint::TransAxis);
         chain.addSegment(Segment(getSegment(parents_chain_root[s+1])->second.segment.getName(),
                                  jnt, f_tip, getSegment(parents_chain_root[s+1])->second.segment.getInertia()));
     }


--- tests/kinfamtest.cpp	2009-08-06 02:37:29.000000000 -0700
+++ tests/kinfamtest.cpp	2010-07-26 13:30:56.822183616 -0700
@@ -220,7 +220,7 @@
     JntArray jnt2(extract_chain2.getNrOfJoints());
     for (int i=0; i<(int)extract_chain2.getNrOfJoints(); i++){
       jnt1(i) = (i+1)*2;
-      jnt2((int)extract_chain2.getNrOfJoints()-i-1) = -jnt1(i);
+      jnt2((int)extract_chain2.getNrOfJoints()-i-1) = jnt1(i);
     }
     solver1.JntToCart(jnt1, f1);
     solver2.JntToCart(jnt2, f2);
